llvm/include/llvm/CAS/OnDiskGraphDB.h

```diff
@@ -309,6 +309,11 @@ class OnDiskGraphDB {
 
   void print(raw_ostream &OS) const;
 
+  /// Hashing function type for validation.
+  using HashingFuncT = function_ref<std::vector<uint8_t>(
```

Comment:
Nit: probably faster to hash into a `SmallVector &` parameter instead of returning a vector, especially since you will do it many times.